src/instructlab/model/list.py

```diff
@@ -0,0 +1,111 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+import logging
+import os
+import time
+
+# Third Party
+from huggingface_hub import ModelCard
+from prettytable import ALL, PrettyTable
+from requests import exceptions as requests_exceptions
+import click
+import pandas
+
+# First Party
+from instructlab import clickext
+from instructlab.configuration import DEFAULTS
+from instructlab.model.backends.backends import is_model_gguf
+
+logger = logging.getLogger(__name__)
+
+
+@click.command(name="list")
+@clickext.display_params
+@click.option(
+    "--model-dirs",
+    help="Base directories where models are stored.",
+    multiple=True,
+    default=[DEFAULTS.MODELS_DIR],
+    show_default=True,
+)
+@click.option(
+    "--list-checkpoints",
+    type=click.BOOL,
+    default=False,
+    is_flag=True,
+    show_default=True,
+)
+def model_list(model_dirs: list[str], list_checkpoints: bool):
+    """lists models"""
+    t = PrettyTable(["Model Name", "Model Age", "Model Size"], hrules=ALL)
+    for dir in model_dirs:
+        for entry in Path(dir).iterdir():
+            # if file, just tally the size. This must be a GGUF.
+            if entry.is_file() and is_model_gguf(entry):
+                # stat the gguf, add it to the table
+                stat = Path(entry.absolute()).stat(follow_symlinks=False)
+                f_size = stat.st_size
+                f_size, magnitude = convert_bytes_to_proper_mag(f_size)
+                # add to table
+                age = pandas.to_timedelta(time.time_ns() - stat.st_ctime_ns)
+                t.add_row([entry.name, age, f"{f_size} {magnitude}"])
+            elif entry.is_dir():
+                actual_model_name = ""
```

Comment:
It seems like there's a lot happening in this `elif` branch and it's leading to a lot of visual clutter. I recommend moving this out to a secondary function since it contains a nested for loop w/ a try/catch.

Reply:
That's an excellent suggestion. I also recommend to move if/elif block into a separate private function that returns name, age, and size. It may even a good idea to move each analyze code into its own private function, too. That would make it easier to write unit tests for each implementation.

```python
AnalyzeResult = tuple[str, str, str]
def _analyze_gguf(entry: pathlib.Path) -> AnalyzeResult:
    ...
```